### PR TITLE
Reissuance - additonal error handling (EXPOSUREAPP-11618)

### PR DESCRIPTION
**Pull request by BMItr**
This PR addresses [Exposureapp-11618.](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-12156)

Note:
TechSpec will be updated soon. The property is called **`"error"` and not `"errorCode"`**

### CODE REVIEW

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/core/server/data/DccReissuanceErrorResponse.kt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package de.rki.coronawarnapp.dccreissuance.core.server.data
+
+import com.google.gson.annotations.SerializedName
+
+data class DccReissuanceErrorResponse(
+    @SerializedName("error") val error: String,
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/core/server/data/DccReissuanceErrorResponse.kt:6]: `errorCode` as per [specs]( https://github.com/corona-warn-app/cwa-app-tech-spec/pull/154/commits/e0302a180bfa9aad8693f3835171d711f3318898#diff-b503751806bd910ffe10c62a514f15f69ed3be758d95ecff6e8700e1cab5b5cbR183)

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/core/server/DccReissuanceServer.kt [hunk 2]
```diff
@@ -61,16 +62,24 @@ class DccReissuanceServer @Inject constructor(
...
-            400 -> ErrorCode.DCC_RI_400
-            401 -> ErrorCode.DCC_RI_401
-            403 -> ErrorCode.DCC_RI_403
-            406 -> ErrorCode.DCC_RI_406
-            429 -> ErrorCode.DCC_RI_429
-            500 -> ErrorCode.DCC_RI_500
-            in 400..499 -> ErrorCode.DCC_RI_CLIENT_ERR
-            else -> ErrorCode.DCC_RI_SERVER_ERR
-        }.also { throw DccReissuanceException(errorCode = it) }
+        val serverErrorCode = tryGetServerErrorCode()
```
Comment (mtwalli) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/core/server/DccReissuanceServer.kt:65]: I think better naming would be `val severError = tryGetServerError()` ?
Here we are not just getting the code , but the whole `error (code + message)`

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/core/server/data/DccReissuanceErrorResponse.kt (hunk 1 above)
Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/core/server/data/DccReissuanceErrorResponse.kt:6]: small unit test for parsing the error would be appreciated

Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/core/server/data/DccReissuanceErrorResponse.kt:6]: Please read the description.
There is a missmatch between the errorCodes & and the Codes in the Spec. Max will provide an updated one and then the spec will match to this PR.

Comment (mtwalli) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/core/server/data/DccReissuanceErrorResponse.kt:6]: Ok 👍

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/core/server/DccReissuanceServer.kt (hunk 2 above)
Comment (BMItr) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/dccreissuance/core/server/DccReissuanceServer.kt]: 👌 adjusted

**Review by mtwalli — APPROVED** [review]

**Comment by mtwalli** [comment]
Thx!

**Review by chiljamgossow — APPROVED** [review]
LGTM

**Pull request merged**